src/rp2_common/pico_cyw43_driver/cyw43_bus_pio_spi.c

```diff
@@ -231,7 +231,7 @@ int cyw43_spi_transfer(cyw43_int_t *self, const uint8_t *tx, size_t tx_length, u
         return CYW43_FAIL_FAST_CHECK(-CYW43_EINVAL);
     }
 
-    bus_data_t *bus_data = (bus_data_t *)self->bus_data;
+    volatile bus_data_t *bus_data = (bus_data_t *)self->bus_data;
```

Comment:
Why is this needed?

Reply:
In some of the linked issues, it's mentioned that different compiler optimisation levels and/or different compiler versions can cause the bug to appear / not appear.

> That's odd for a stack variable.

Perhaps it means that this bug goes a bit deeper?

Reply:
Yes, it fails without it. Something fishy is going on in release

Reply:
If I add __attribute((noinline)) to the function it's ok (volatile not needed).

Reply:
I think this patch is still unsound because it checks things in the wrong order (see this comment https://github.com/raspberrypi/pico-sdk/pull/2209#issuecomment-2615466462).

It's likely that pushing the code around like this is adding delays and hiding symptoms.